source/common/tcp_proxy/tcp_proxy.h

```diff
@@ -61,7 +61,7 @@ class UpstreamDrainManager;
 /**
  * Route is an individual resolved route for a connection.
  */
-class Route {
+class Route : public Router::RouteEntry {
```

Comment:
I don't think this part makes sense, as we have tons of not implemented APIs for TCP. I would recommend introducing a common base class that might be shared between TCP and HTTP routes (and maybe other procols in the future).

Reply:
I think FilterChain is a more appropriate abstraction for HTTP route IMHO.